src/hooks/useAxiosWithAuth0.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react';
+import axios from 'axios';
+import { useAuth0 } from '@auth0/auth0-react';
+
+/*
+NOTE:
+React requires that we can only call hooks in function components or within other hooks. As axiosWithAuth is simply a helper function we've built a useAxiosWithAuth0() hook in which we've called both axiosWithAuth AND useAuth0().
+axiosWithAuth serves the same function as it typically does. useAuth0() gives access to getAccessTokenSilently() for authentication/authorization purposes. 
+This hook below is exported and used throughout the app in place of axiosWithAuth.
+THIS useAxiosWithAuth0 HOOK SIMPLY SERVES TO HELP CONNECT THE AXIOS CALL AND THE AUTHENTICATION.
+*/
```

Comment:
Loving this living documentation in what is essentially an extra-specialized `axiosWithAuth` - and great work turning this into a handy hook, too!